src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -0,0 +1,995 @@
+                    values,
+                    ValueMappingTransformer.KeyColumnName,
+                    ValueMappingTransformer.ValueColumnName,
+                    treatValuesAsKeyValue);
+
+        // Handler for vector value types
+        private static IDataView ConvertToDataView(IHostEnvironment env, IEnumerable<TKey> keys, IEnumerable<TValue[]> values)
+            => DataViewHelper.CreateDataView(env, keys, values, ValueMappingTransformer.KeyColumnName, ValueMappingTransformer.ValueColumnName);
+    }
+
+    public class ValueMappingTransformer : OneToOneTransformerBase
+    {
+        internal const string Summary = "Maps text values columns to new columns using a map dataset.";
+        internal const string LoaderSignature = "ValueMappingTransformer";
+        internal const string UserName = "Value Mapping Transform";
+        internal const string ShortName = "ValueMap";
+
+        internal const string TermLookupLoaderSignature = "TermLookupTransform";
+
+        // Stream names for the binary idv streams.
+        private const string DefaultMapName = "DefaultMap.idv";
+        protected static string KeyColumnName = "Key";
+        protected static string ValueColumnName = "Value";
+        private ValueMap _valueMap;
+        private Schema.Metadata _valueMetadata;
+        private byte[] _dataView;
+
+        public ColumnType ValueColumnType => _valueMap.ValueType;
+        public Schema.Metadata ValueColumnMetadata => _valueMetadata;
+
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "VALUMAPG",
+                verWrittenCur: 0x00010001, // Initial.
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(ValueMappingTransformer).Assembly.FullName);
+        }
+
+        private static VersionInfo GetTermLookupVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "TXTLOOKT",
+                // verWrittenCur: 0x00010001, // Initial.
+                verWrittenCur: 0x00010002, // Dropped sizeof(Float).
+                verReadableCur: 0x00010002,
+                verWeCanReadBack: 0x00010002,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(ValueMappingTransformer).Assembly.FullName);
+        }
+
+        public sealed class Column : OneToOneColumn
+        {
+            public static Column Parse(string str)
+            {
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            public bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        public sealed class Arguments
```

Comment:
>Arguments [](start = 28, length = 9)

this class is mostly for CMD, so either leave it as it was in TermLookup, or provide aliases with old names (you can use ShortName="AliasA,AliasB,AliasC,"

Reply:
I have added aliases.

---
In reply to: [243463898](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243463898) [](ancestors = 243463898)